pennylane/operation.py

```diff
@@ -798,13 +798,15 @@ def id(self):
     def name(self, value):
         self._name = value
 
-    def label(self, decimals=None, base_label=None):
+    def label(self, decimals=None, base_label=None, cache=None):
```

Comment:
This is a bit unfortunate, is it possible to simplify this change by moving the cache logic out of the label method and keeping it in the drawer? I think that's how the previous drawer handled it, or are there arguments against this?

Reply:
When we add a new class, like a new operation, we want to limit the number of places in the codebase we have to change.  Previously, adding any new operation forced the `RepresentaitonResolver` class in the drawer module to be changed as well.  That class was mostly "if this class type, then this" special cases over and over again. That made it difficult to understand and maintain.

Adding and updating operations is something we do all the time.  Updating the drawer module was something that rarely happened. This led to them being out of sync with each other.

So now Operators decide how they want themselves to be drawn.  This allows us to easily handle special cases like `Projector`.

I do like the addition of the `cache` dictionary because it will be easily extensible in the future.  Right now it only stores matrix and nested tape information, but nothing is stopping any operation from taking advantage of it in the future for other types of information.

Is there something in particular that bothers you about adding the `cache` dictionary? There may be another solution.  Well, a solution other than just ignoring matrices.

Reply:
Okay fair enough! I was thinking that the cache logic fits better in the drawer because the specifics of how a matrix is represented is merely dependant on the drawing context and independent of the actual operation itself. This part of the drawing logic should work for all operations no, so that it wouldn't need updating when new operations are added?

But you make some good points as well so I'd fine with this approach too.